setup_lacework_agent.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+set -e
+
+LACEWORK_INSTALL_PATH="{{ LaceworkInstallPath }}"
+
+# TODO: Fetch the token from AWS SSM Parameter Store instead of taking it in as a Command parameter (avoid leaks in the AWS Console)
+TOKEN="{{ Token }}"
+TAGS="{{ Tags }}"
+
+
+# TODO: Handle systems that don't have systemctl
+if systemctl is-active --quiet kubelet; then
+  echo "This host appears to be a Kubernetes node, please use the Kubernetes deployment method (https://support.lacework.com/hc/en-us/articles/360005263034-Deploy-on-Kubernetes)."
+  exit 0
+fi
+
+if [ ! -d "$LACEWORK_INSTALL_PATH" ]; then
+  echo "Lacework agent not installed, installing..."
+
+  # TODO: Add the support for hosts that don't have curl installed
+  curl https://packages.lacework.net/install.sh > /tmp/install.sh
+
+  chmod +x /tmp/install.sh
+
+  # TODO: Pass tags to the installation script
+  sudo /tmp/install.sh "$TOKEN"
+
+  rm /tmp/install.sh
+else
+  echo "Lacework is already installed, configuring the agent token..."
```

Comment:
Here, I expect the `install.sh` script can handle the configuration.

So the first code path calls `install.sh` that does the config. The second code path is for instances where Lacework was pre-installed (using Ansible in EC2 Image Builder for example) and where all we need is to setup the `config.json` file.

We can indeed close the `if` and chose to always write `config.json` file, therefore overriding what was done by `install.sh`.

What do you prefer? You know the `install.sh` script way better than I do.

Reply:
I would suggest you close the if so that it only detects if it is installed or not. Then add a new if to modify the configuration only if the user provided `$TAGS`.

Reply:
> Then add a new if to modify the configuration only if the user provided $TAGS.

So what happens for the Lacework agent that was pre-installed on an AMI using Ansible?

We need to configure the Token for those. This is why in my mind, we simply drop the if for the config.json file setup.

Reply:
Great point. To be super honest, the more I think bout this module/document, the more I feel we should **not** design it thinking on existing installations. If this module runs on a system that already has the Lacework Agent installed, we should not modify it. Otherwise, we will be codifying many scenarios that we support, like you said, Ansible, k8s, etc.

Lets rename this document from `setup-lacework-agent` to `install-lacework-agent` so that it is clear that this command will only install the agent and nothing else. Thoughts?

Reply:
I improved this part based on the discussion we had in Slack!